framework/source/class/qx/Promise.js

```diff
@@ -517,6 +517,35 @@ qx.Class.define("qx.Promise", {
       return promise;
     },
 
+    /**
+     * Returns a promise that resolves when all of the promises in the object properties have resolved, 
+     * or rejects with the reason of the first passed promise that rejects.  The result of each property
+     * is placed back in the object, replacing the promise. 
+     * 
+     * @param value {var} An object
+     * @return {qx.Promise}
+     */
+    allOf: function(value) {
+      function action(value) {
+        var arr = [];
+        var names = [];
+        for (var name in value) {
+          if (value.hasOwnProperty(name) && value[name] instanceof qx.Promise) {
+            arr.push(value[name]);
+            names.push(name);
+          }
```

Comment:
this will silently ignore stuff non-promises ... at least in debug mode, there might be a warning warranted ...

Reply:
that's deliberate - the `qx.Promise.all` does the same, in that non-promises are not changed, the intention is that the property values (or in `.all`'s case, the array elements) are transformed so that there are no Promises left.

Reply:
ok ... maybe add a note to the documentaion